flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -323,10 +323,6 @@ public CheckpointCoordinator(
 
         try {
             this.checkpointStorageView = checkpointStorage.createCheckpointStorage(job);
-
-            if (isPeriodicCheckpointingConfigured()) {
-                checkpointStorageView.initializeBaseLocationsForCheckpoint();
-            }
```

Comment:
? Are you reverting FLINK-23180? Commit message is also strange.

Reply:
I am not reverting it. But I am implementing it in a different way. See the discussion https://github.com/apache/flink/pull/17151#issuecomment-920708455 People involved in the other ticket did not want to wait for this solution.

Reply:
Why is the commit message strange?

Reply:
Secondly, now I guess we will be failing a bit later and first checkpoint can theoretically take longer as a result of this change?

Would it be complicated to initialise base `initializeBaseLocationsForCheckpoint()` if `isPeriodicCheckpointingConfigured()`, but if `isPeriodicCheckpointingConfigured()` is false, initialise it also in the `FsCheckpointStorageAccess#initializeLocationForCheckpoint()` the way you are doing it? It would definitely complicate the system by having two ways of initialisation, so I'm not yet convinced which option is better.

Reply:
No, it would not be complicated. It would be very straightforward to do.

In general the change that introduced `initializeBaseLocationsForCheckpoint` was not about falling early, but so that we call the filesystem just once, which still holds. But I agree the current code might introduce some delay to the first checkpointing compared to master.

Reply:
Given the minimal scope of the required changes let's do it twofold.